### PR TITLE
IA-3123: Review the look and feel of permissions list (with Read/Write)

### PR DESCRIPTION
**Pull request by hakifran**
Explain what problem this PR is resolving
- Review the look and feel of permissions list (with Read/Write)
Related JIRA tickets : [IA-3123](https://bluesquare.atlassian.net/browse/IA-3123)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc
- `iaso/docs/pages/dev/how_to/add_new_permission/add_new_permission.md`
- #### 4. If the added permission must be coupled with another permission like read and edit

## Changes

- Added a READ_EDIT_PERMISSIONS constant in hat/menupermissions/constants.py
- Format Permissions to include the matching read-edit
- Display the matching read-edit permission
- Replace Switch component by checkbox
- Add translations

## How to test
- Go Admin ---> Users ---> Management and Go to Admin ---> User Roles
- Create or edit a user and a user role
- Add and remove permissions and specifically those with read and edi permission

## Print screen / video
[Screencast from 2024-10-02 12-36-51.webm](https://github.com/user-attachments/assets/29990eaf-960f-4c9a-ac6c-590f610dbfeb)
[Screencast from 2024-10-02 12-37-22.webm](https://github.com/user-attachments/assets/567d11f1-5b55-4be4-8e15-1b00232c45a4)

[IA-3123]: https://bluesquare.atlassian.net/browse/IA-3123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/pages/dev/how_to/add_new_permission/add_new_permission.md [hunk 1]
```diff
@@ -21,21 +21,29 @@
...
 
-## 4. Make and run migration
+## 4. If the added permission must be coupled with another permission like read and edit
+- Go to `/hat/menupermissions/constants.py`
+- Add the added permission and it's related permission as an item of the `READ_EDIT_PERMISSIONS` dictionnary
+- The item should have a key which reprensente the string name which will be displayed
+- The item should have a dictionnary reprensenting the coupled permissions, the keys (should be two keys) are `read` and `edit` or other keys like `no-admin` and `admin`
+- The item should look like `item_key": {"read": "added_permission", "edit": "coupled_permission"}`
+- Add translations for all the keys(`item_key, read and edit`) and the tooltip message of the principal key(`item_key`)
+
```
Comment (beygorghor) [inline docs/pages/dev/how_to/add_new_permission/add_new_permission.md:31]: In order to display a new permission in the dashboard you need to add it to PERMISSIONS_PRESENTATION too no ?

File: hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx:1]: PermissionCheckBoxes

File: hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    type: string;
+};
+
+const PermissionCheckBoxs: React.FunctionComponent<Props> = ({
+    value,
+    codeName,
+    settings,
+    setPermissions,
+    permissions,
+    type,
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx]: This is never used

File: hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}) => {
+    const { original } = settings.row;
+
+    const handleCheckboxChange = (
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/PermissionCheckBoxs.tsx]: useCallback can be nice here

File: hat/assets/js/apps/Iaso/domains/users/config.js [hunk 4]
```diff
@@ -180,16 +184,59 @@ export const useUserPermissionColumns = ({ setPermissions, currentUser }) => {
                 width: 50,
                 Cell: settings => {
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/config.js:179]: Thie cell is too big, we need to create another component aside for this

File: hat/assets/js/apps/Iaso/domains/users/config.js [hunk 5]
```diff
@@ -180,16 +184,59 @@ export const useUserPermissionColumns = ({ setPermissions, currentUser }) => {
...
+                                    style={{
+                                        display: 'inline-flex',
+                                        gap: '5px',
+                                    }}
+                                >
+                                    {permissions.map(([key, value]) => {
+                                        const hasPermission =
+                                            role.permissions.includes(value);
+                                        return (
+                                            <Stack
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/config.js]: never used this, well done ❤️

**Review by beygorghor — CHANGES_REQUESTED** [review]
Good job, We will improve lay-out a bit after. I'm just wondering why you have read twice in the checkboxes? it should be read and write no ?

Screenshot 2024-10-02 at 10 46 39

**Comment by hakifran** [comment]
> Good job, We will improve lay-out a bit after. I'm just wondering why you have read twice in the checkboxes? it should be read and write no ?
> Screenshot 2024-10-02 at 10 46 39

Yeah it's a translation issue: `"iaso.permissions.readEdit.edit": "Read",`. I Will fix that

File: docs/pages/dev/how_to/add_new_permission/add_new_permission.md (hunk 1 above)
Comment (hakifran) [reply docs/pages/dev/how_to/add_new_permission/add_new_permission.md:31]: It's done on third point before the added description(4th)

**Review by beygorghor — APPROVED** [review]
I have updated your branch with main and resolve conflicts. I have tested it after and works like a charm

**Pull request merged**